docs/preview/03-Features/secret-store/create-new-secret-provider.md

```diff
@@ -165,46 +165,47 @@ namespace Application.Controllers
 }
 ```
 
-### Adding secret versions to your secret provider
-When your secret storage location supports versioned secrets, you could consider adapting your secret provider to support these.
-For more information on how you can use multiple versions of a secret in your application, see [this dedicated page](./versioned-secret-provider.md).
+For more information on consuming cached secrets, see [this dedicated page on consuming secrets](../secrets/general.md).
 
-Implement from `IVersionedSecretProvider` instead of `ISecretProvider` to allow the secret store to pick that your secret provider supports secret versions.
-The following example shows how the registry secret provider only supports two versions of a secret:
+### Adding synchronous secret retrieval to your secret provider
+When you want to use your secret provider in synchronous environments, you can implement the `ISyncSecretProvider` instead (implement also `ISecretProvider`).
```

Comment:
What is the reason why this is done with a separate interface ?
I don't see any benefit for that ?  Don't you think it becomes more complex to use this ? 

The issue here is, that you -as consumer- are obliged to know if you want to inject an ISecretProvider or an ISyncSecretProvider, and this breaks imho the abstraction.

Reply:
If we would alter the `ISecretProvider`, we would not only break contract, but we would also force any implementation to be synchronous. Our HashiCorp Vault, for example, cannot do that, and I imagine other implementations that are not able to do that.
Which is already the same conclusion you came to below this one 😉 .